src/sentry/rules/conditions/tagged_event.py

```diff
@@ -47,52 +51,66 @@ class TaggedEventCondition(EventCondition):
     def passes(self, event, state, **kwargs):
         key = self.get_option("key")
         match = self.get_option("match")
-        value = self.get_option("value")
+        value = self.get_option("value", "")
 
-        if not (key and match and value):
+        if not (key and match):
             return False
 
         value = value.lower()
         key = key.lower()
 
-        tags = (
+        tags = (k.lower() for k, v in event.tags)
+
+        values = (
             v.lower()
             for k, v in event.tags
             if k.lower() == key or tagstore.get_standardized_key(k) == key
         )
 
-        if match == MatchType.EQUAL:
-            for t_value in tags:
+        if match == MatchType.IS_SET:
+            for tag in tags:
+                if tag == key:
+                    return True
+            return False
+
+        elif match == MatchType.NOT_SET:
+            for tag in tags:
+                if tag == key:
+                    return False
+            return True
+
```

Comment:
Just to be cautious, how about we add 
```
if not value:
    return False
```
here after we check `IS_SET` and `NOT_SET`? We never used to compare an empty value to existing tag values, just concerned it might change behaviour in some edge cases.

Reply:
Fair enough :)